src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -308,12 +309,14 @@ public static double log10sumLog10(final double[] log10p, final int start, final
 
 
     /**
-     * A helper class to maintain a cache of log values.
-     * The cache is immutable after creation.
+     * A helper class to maintain a cache of log10 values.
+     * The cache expands when a number is not available.
+     * NOTE: this cache is NOT thread safe and it must not be accessed from multiple threads.
      */
     private static final class Log10Cache {
```

Comment:
actually, i think the gatk3 code is better than doing the thread local (which requires a map lookup for every call). I'll put the gatk3 version in for the second review